In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_excel('anova_lab_data.xlsx')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Power         15 non-null     object 
 1   Etching Rate  15 non-null     float64
dtypes: float64(1), object(1)
memory usage: 368.0+ bytes


In [12]:
df. columns= df. columns. str. lower()

In [13]:
df

power   etching rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [32]:
df.columns=[h.lower().replace(' ','_') for h in df.columns]

In [33]:
df

power_  etching_rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [34]:
import statsmodels.api as sm

H0; there is no difference in the mean etching rate for different levels of power

HA; there is difference in the mean etching rate for different levels of power

In [35]:
# Ordinary Least Squares (OLS) model

In [36]:
from statsmodels.formula.api import ols

In [38]:
model=ols('etching_rate ~ C(power_)',data=df).fit()

In [40]:
results_table=sm.stats.anova_lm(model)

In [41]:
results_table

df     sum_sq   mean_sq          F    PR(>F)
C(power_)   2.0  18.176653  9.088327  36.878955  0.000008
Residual   12.0   2.957240  0.246437        NaN       NaN

What is the significance level?
The significance level of 0.05

What are the degrees of freedom of model, error terms, and total DoF?
Deegres of freedom are 2 in between groups
Total DoF is 14
The F-statistic= 36.878955 and the p-value= 0.000008


Conclusions
Based on p value we can reject H0 and say that overall there is a significant difference 
in mean etching rate depending on different levels of power